lib/main.js

```diff
@@ -292,15 +319,13 @@ class VM extends EventEmitter {
 		const script = code instanceof VMScript ? code : new VMScript(code);
 		script._compileVM();
 
-		try {
-			return this._internal.Decontextify.value(script._compiledVM.runInContext(this._context, {
-				filename: script.filename,
-				displayErrors: false,
-				timeout: this.options.timeout
-			}));
-		} catch (e) {
-			throw this._internal.Decontextify.value(e);
-		}
+		return doWithTimeout(()=>{
```

Comment:
We should use `doWithTimeout` only when timeout is specified. Related to my previous comment.